### PR TITLE
Bug 1901301: Handle case when Provisioning CR is absent on BareMetal Platform

### PR DESCRIPTION
**Pull request by sadasu**

### CODE REVIEW

File: controllers/provisioning_controller.go [hunk 1]
```diff
@@ -270,6 +271,15 @@ func (r *ProvisioningReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 
 // SetupWithManager configures the manager to run the controller
 func (r *ProvisioningReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	namespacedName := types.NamespacedName{Name: BaremetalProvisioningCR}
+	baremetalConfig, err := r.readProvisioningCR(namespacedName)
+	if err != nil || baremetalConfig == nil {
+		err = r.updateCOStatus(ReasonComplete, "nothing to do in assisted install", "")
```
Comment (stbenjam) [inline controllers/provisioning_controller.go:277]: I think we want to not mention assisted. Conceivably, someone could install a compact 3-node baremetal cluster without provisioning services by deleting the CR.

So maybe something like "Provisioning CR not found on baremetal platform; marking operator as available"?

Comment (sadasu) [reply controllers/provisioning_controller.go:277]: I did not like that string. Thanks for the suggestion.

**Comment by dhellmann** [comment]
This is going to conflict with #69. I don't mind rebasing, but we could also combine them into 1 PR if I rebase this on top of the other. Let me know what you think.

**Comment by sadasu** [comment]
> This is going to conflict with #69. I don't mind rebasing, but we could also combine them into 1 PR if I rebase this on top of the other. Let me know what you think.

Yes, it will. Lets combine this into 1 PR (#69) as soon as we use this PR to iron out our CI issues. (I didn't want to add more noise to #69.)

**Comment by stbenjam** [comment]
> > This is going to conflict with #69. I don't mind rebasing, but we could also combine them into 1 PR if I rebase this on top of the other. Let me know what you think.
> 
> Yes, it will. Lets combine this into 1 PR (#69) as soon as we use this PR to iron out our CI issues. (I didn't want to add more noise to #69.)

Is there any chance we could land this one as-is? I'd prefer to unblock assisted as quickly as possible as some other work is held up on it.

**Comment by stbenjam** [comment]
Failure looks related:

```
oc explain should contain proper spec+status for CRDs [Suite:openshift/conformance/parallel] expand_less | 36s
-- | --
fail [github.com/openshift/origin/test/extended/cli/explain.go:442]: Unexpected error:     <*errors.errorString \| 0xc002860a50>: {         s: "oc explain [\"provisionings\" \"--api-version=metal3.io/v1alpha1\"] result {KIND:     Provisioning\nVERSION:  metal3.io/v1alpha1\n\nDESCRIPTION:\n     <empty>} doesn't match pattern {(?s)DESCRIPTION:.*FIELDS:.*spec.*<.*>.*(status.*<.*>.*)?}",     }     oc explain ["provisionings" "--api-version=metal3.io/v1alpha1"] result {KIND:     Provisioning     VERSION:  metal3.io/v1alpha1
```

**Comment by romfreiman** [comment]
cc: @hexfusion @YuviGold . assisted-installer passed here

**Comment by stbenjam** [comment]
/retitle Bug 1901301: Handle case when Provisioning CR is absent on BareMetal Platform

**Pull request merged**